hazelcast/src/hazelcast/client/connection/InSelector.cpp

```diff
@@ -66,7 +66,7 @@ namespace hazelcast {
                             sleepingSocket->receive(&wakeUpSignal, sizeof(int));
                         } else {
                             boost::shared_ptr<Connection> conn = connectionManager.getConnectionIfAvailable(fd);
-                            if (conn.get() != NULL) {
+                            if (conn.get() != NULL && conn->live) {
```

Comment:
OK, I understand now. i think it is not EBADF but errorfds will be filled for the close case (if the fd was initally valid (i.e. open socket) and closed during select wait). No need for wakeup.